src/main/java/org/wiremock/integrations/testcontainers/WireMockContainer.java

```diff
@@ -65,7 +65,7 @@ public class WireMockContainer extends GenericContainer<WireMockContainer> {
     private final StringBuilder wireMockArgs;
     private final Map<String, Stub> mappingStubs = new HashMap<>();
     private final Map<String, MountableFile> mappingFiles = new HashMap<>();
-    private final Map<String, Extension> extensions = new HashMap<>();
+    private final Map<String, WireMockExtension> extensions = new HashMap<>();
```

Comment:
Might be very subjective, but for me WiremockExtention class looks a bit redundant.
I would distinguish classNames and jarFIles to simplify usage
 
```java
private final Set<String> extensionClassNames = new LinkedHashSet<>();
private final Set<File> extensionJars = new LinkedHashSet<>();
```

Also i would simplify methods, this three could solve almost any case
```java
withExtention(String className, File jar) // 1
withExtention(Collection<String> classNames, File jar) // 2
withExtention(Collection<String> classNames, Path directoryWithJars) // 3
```

Or we could make it even more simple

```java
withExtentionJar(File jar)
withExtentionFolder(Path folderWithJars)
withExtentionClassName(String className)
```

Reply:
This class has a package visibility scope, so it is okay IMHO. I also resorted to a single class name, because this is how the extensions I actually defined in the main WireMock codebase and docs. So fun and extension is a single class, and there is no grouping of extensions (e.g."plugin") one could understand under this definition

I'd like to speak to @tomakehurst to see you weather report fix this terminology and approach accordingly